lib/internal/streams/promises.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const {
+  Promise,
+} = primordials;
+
+function pipeline() {
+  return new Promise((resolve, reject) => {
+    require('internal/streams/pipeline')(...arguments, (err, value) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(value);
+      }
+    });
+  });
+}
+
+function finished(stream, opts) {
+  return new Promise((resolve, reject) => {
+    require('internal/streams/end-of-stream')(stream, opts, (err) => {
```

Comment:
I think the require here should be cached, i.e.

if (!eos) eos =  require('internal/streams/end-of-stream');
eos(...);

ditto above

Reply:
also doing the require outside of the promise might make it more inlineable?

Reply:
fixed